bin/gaffer.cmd

```diff
@@ -0,0 +1,171 @@
+@echo off
+
+setlocal EnableDelayedExpansion
+
+set GAFFER_ROOT=%~dp0%..
+
+set HOME=%USERPROFILE%
+
+set GAFFER_JEMALLOC=0
+
+call :prependToPath "%GAFFER_ROOT%\glsl" IECOREGL_SHADER_PATHS
+call :prependToPath "%GAFFER_ROOT%\glsl" IECOREGL_SHADER_INCLUDE_PATHS
+
+call :prependToPath "%GAFFER_ROOT%\fonts" IECORE_FONT_PATHS
+call :prependToPath "%GAFFER_ROOT%\ops" IECORE_OP_PATHS
+
+call :prependToPath "%USERPROFILE%\gaffer\opPresets;%GAFFER_ROOT%\opPresets" IECORE_OP_PRESET_PATHS
+call :prependToPath "%USERPROFILE%\gaffer\procedurals;%GAFFER_ROOT%\procedurals" IECORE_PROCEDURAL_PATHS
+call :prependToPath "%USERPROFILE%\gaffer\proceduralPresets;%GAFFER_ROOT%\proceduralPresets" IECORE_PROCEDURAL_PRESET_PATHS
+
+set CORTEX_POINTDISTRIBUTION_TILESET=%GAFFER_ROOT%\resources\cortex\tileset_2048.dat
+
+call :prependToPath "%USERPROFILE%\gaffer\apps;%GAFFER_ROOT%\apps" GAFFER_APP_PATHS
+
+call :prependToPath "%USERPROFILE%\gaffer\startup" GAFFER_STARTUP_PATHS
+call :appendToPath "%GAFFER_ROOT%\startup" GAFFER_STARTUP_PATHS
+
+call :prependToPath "%GAFFER_ROOT%\graphics" GAFFERUI_IMAGE_PATHS
```

Comment:
> I'm completely unqualified to review any of this.

I'm envious! Window batch scripting is the most unholy "language" I've come across and I don't wish it on anyone.

I'll give a try to conditionally installing a Linux or Windows version of `gaffer` and see if that works.

I'm not sold on using the bash wrapper on Windows because it would require a system component be installed that isn't by default. That would mean someone just trying Gaffer would have to install additional components, and might not be possible at all if you don't have admin privileges.

How feasible would it be to move to a Python based environment configurer? I suspect that would be pretty disruptive to current users.

Reply:
> I'm not sold on using the bash wrapper on Windows because it would require a system component be installed that isn't by default.

Cool, makes sense.

> How feasible would it be to move to a Python based environment configurer?

It's probably doable. Is Python a guaranteed presence on Windows though? If not, we might need a pre-wrapper to make sure Gaffer's Python is available before we can run the python-based wrapper...one for another time...

Reply:
Right, I had already been thinking we'd use the pre-wrapper, but didn't explain that well. Python is not a guaranteed presence on Windows so we'd need to use our own.